src/disassembler.rs

```diff
@@ -0,0 +1,17 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fmt;
+
+// Supported architectures
+enum Architecture {
+    X86,
+    X86_64,
+}
+
```

Comment:
You're going to need conversion from `Architecture` to the Capstone `cs_arch` enum, so I'd implement that via the `From` trait:

```
impl From<capstone::cs_arch> for Architecture {
    fn from(arch: capstone::cs_arch) -> Self {
        ...
    }
}
```

That'll automatically implement the `Into` trait as well and then when you have an `Architecture` value but need the Capstone value, you can just call `arch.into()` and all will be well.